### PR TITLE
when properties use the same default they use the same object in memory completes #90

### PR DESCRIPTION
**Pull request by seancookr**
Recently I noticed that if a default value is specified for a property in a class inheriting from `Dolly::Document`, that default property and it's default properties are tied to class variable. and that object itself is never duplicated, thus each time you touch the object you touch the same object in memory.  This issues was causing classes to be instantiated with values that were never intended (if other instances of the class had modified their property) (this is not a problem with singletons).

This PR duplicated that default object so that it is a new object in memory.

### CODE REVIEW

File: test/document_test.rb [hunk 1]
```diff
@@ -433,6 +433,16 @@ def setup
...
   end
 
+  test 'default properties do not update the class default properties' do
+    doc = DocWithSameDefaults.new
+    assert_equal [], DocWithSameDefaults.properties[:foo].default
+    assert doc.foo.push 'foo'
+    assert_equal ['foo'], doc.foo
+    assert_equal [], DocWithSameDefaults.properties[:foo].default
+    assert second_doc = DocWithSameDefaults.new
+    assert second_doc.foo.empty?
```
Comment (javierg) [inline test/document_test.rb]: consider checking againts `[]` so we avoid false positives if is `nil`

Comment (seancookr) [inline test/document_test.rb]: Done

File: test/document_test.rb [hunk 2]
```diff
@@ -433,6 +433,16 @@ def setup
...
   end
 
+  test 'default properties do not update the class default properties' do
+    doc = DocWithSameDefaults.new
+    assert_equal [], DocWithSameDefaults.properties[:foo].default
+    assert doc.foo.push 'foo'
+    assert_equal ['foo'], doc.foo
+    assert_equal [], DocWithSameDefaults.properties[:foo].default
+    assert second_doc = DocWithSameDefaults.new
+    assert_equal [], second_doc.foo
```
Comment (javierg) [inline test/document_test.rb]: :clap:

**Comment by javierg** [comment]
:shipit:

**Pull request merged**